src/main/java/hudson/plugins/ec2/EC2RetentionStrategy.java

```diff
@@ -164,11 +164,6 @@ private long internalCheck(EC2Computer computer) {
 
             final long idleMilliseconds = this.clock.millis() - computer.getIdleStartMilliseconds();
 
-            //Stopped instance restarted and Idletime has not be reset
-            if ( uptime <  idleMilliseconds) {
```

Comment:
This was added to prevent computers with stale uptimes from terminating.

Reply:
Yes but [this](https://github.com/jenkinsci/ec2-plugin/pull/359) PR should have fixed the stale uptimes, right?

Reply:
Much like this PR that number could possibly be stale due to eventual consistency

Reply:
I am not sure I understand, I am under the impression that #399 should permanently fix the stale uptime issue in `EC2RetentionStrategy.java` since `getstate()` will get the latest info from AWS and refresh uptime value. Could you please elaborate on the concern?

Reply:
I do not know how the EC2 API behaves in this case. Previously we had issues of stale uptimes causing nodes to terminate but those nodes were just spun but did throw the exception seen in this PR

Reply:
@datallah IIRC you added this bit, is it due to this?

Reply:
@res0nance 
Hello Rihaan,
I can remove this block for now if you'd want to merge the other change and maybe discuss this separately. My only concern is that instances which are manually started on AWS will never be stopped nor connected to Jenkins by the ec2 plugin in that scenario. Since, for manually started nodes, `uptime` will always be less than `idleMilliseconds`

Reply:
getIdleMilliseconds returns when the node was started i.e when it was connected to jenkins. That number should never be further into the future than the uptime.

https://github.com/jenkinsci/jenkins/blob/6d6a0a665b4ac0864646d16b114bb78b5f21f9c1/core/src/main/java/hudson/model/Computer.java#L1058
https://github.com/jenkinsci/jenkins/blob/6d6a0a665b4ac0864646d16b114bb78b5f21f9c1/core/src/main/java/hudson/model/Executor.java#L887

Reply:
@res0nance actually, I didn't add that bit, that has been like that for a while.

I agree with the sentiment that the check should no longer be necessary because of the fixes in #359.

Reply:
I tried out an experiment on this scenario and was able to confirm that `uptime < idleMilliseconds` can be reproduced as follows
1. Consider a node that was stopped by ec2-plugin on idle timeout after finishing the build
2. For this node , `uptime >  idleMilliseconds` since uptime is counted from the time when the instance was last started but idle time is calculated after the connection process and build completes.
3. Now, manually start this stopped instance. Since ec2-plugin is not aware of this launch, ec2-plugin will never connect to the node and it will not come online. Idle time here will represent the time since the node got idle before it was stopped by the plugin while uptime will be the time since you just manually launched this machine on AWS. Thus, `uptime < idleMilliseconds`. The presence of this condition in `EC2RetentionStrategy.java` basically prevents the ec2-plugin from ever shutting down that manually started instance unless you manually initiate the connection procedure which in turn resets the idle time.
4. A similar issue occurs in `JENKINS-57795` where ec2-plugin starts the instance but never connects to it, this causes idlemilliseconds to never reset
5. Use the following script in the Jenkins script console to try this out
```groovy
package hudson.plugins.ec2;
import java.util.concurrent.TimeUnit;

import org.apache.commons.lang.math.NumberUtils;

Jenkins jenkins = Jenkins.getInstance()
int STARTUP_TIME_DEFAULT_VALUE = 30;
int STARTUP_TIMEOUT = NumberUtils.toInt(System.getProperty(EC2RetentionStrategy.class.getCanonicalName() + ".startupTimeout",String.valueOf(STARTUP_TIME_DEFAULT_VALUE)), STARTUP_TIME_DEFAULT_VALUE)

EC2Computer computer = jenkins.getComputer("Your node name in jenkins")

EC2RetentionStrategy ret = new EC2RetentionStrategy('-5')

idleTerminationMinutes = ret.idleTerminationMinutes

println computer.isIdle()
println computer.getState();
println computer.isOffline()

long uptime = computer.getUptime()
long idleMilliseconds = System.currentTimeMillis() - computer.getIdleStartMilliseconds()

println uptime < idleMilliseconds
```

Reply:
In light of this new discovery would you re-add that check?

Reply:
No, I meant to explain that removing that check is necessary otherwise it will prevent some slaves from getting stopped even after idle timeout

Reply:
Sorry, I misunderstood

Reply:
that was introduced because the startup time can be longer than idle